src/main/java/org/sonarlint/intellij/analysis/OnTheFlyFindingsHolder.kt

```diff
@@ -67,6 +91,38 @@ class OnTheFlyFindingsHolder(private val project: Project) : FileEditorManagerLi
         }
     }
 
+    private fun findExtensions(): MutableSet<@NlsSafe String?>? {
+        var extensions = project.getOpenFiles().stream()
+            .filter { e ->
+                e.extension.equals("java")
+                    || e.extension.equals("pks")
+                    || e.extension.equals("pkb")
+                    || e.extension.equals("sql")
+                    || e.extension.equals("scala")
+                    || e.extension.equals("swift")
+            }
+            .map { e -> e.extension }
+            .collect(Collectors.toSet())
+        return extensions
+    }
+
+    private fun showAd(notifications: SonarLintProjectNotifications, language: String) {
+        val isDontAskAgain = getGlobalSettings().isDontAskAgain
+        if (!isDontAskAgain) {
+            val lastAdDate = getGlobalSettings().lastAdNotificationDate
+            if (lastAdDate == null) {
+                notifications.notifyWiderLanguageSupport(language)
+                getGlobalSettings().updateLastModifiedDate(Instant.now())
+            } else {
```

Comment:
This if/else-if split is a bit strange, as the action made is the same in both cases. What about grouping like:

> if (lastPromotionDate?.isBefore(Instant.now().minus(Period.ofDays(7))) == true)

(not tested)

Reply:
This check will be needed for other kind of notifications, so you would have to duplicate it. I think it would be nice to at least extract the snippet to a separate `shouldNotify` method, which better shows the intent of the snippet